pkg/xds/sync/dataplane_proxy_builder.go

```diff
@@ -105,6 +108,26 @@ func (p *DataplaneProxyBuilder) resolveRouting(
 		return nil, nil, err
 	}
 
+	var domains []xds.VipDomains
+	outbounds := dataplane.Spec.Networking.Outbound
+	if dataplane.Spec.Networking.GetTransparentProxying() != nil && !dataplane.Spec.IsIngress() {
+		// resolve all the domains
+		domains, outbounds = xds_topology.VIPOutbounds(core_model.MetaToResourceKey(dataplane.Meta), meshContext.Dataplanes.Items, zoneIngresses.Items, dnsResolver.GetVIPs(), dnsResolver.GetDomain(), matchedExternalServices)
+
+		// Update the outbound of the dataplane with the vips
```

Comment:
I believe this is preserving the non vip outbounds (In case a user defined outbounds in their dataplane resource).
What we do though is making sure we're not duplicating the outbounds in case there are still the generated ones in the resource.
WDYT?

Reply:
oh right, I forgot about non-vip outbounds on k8s :)